src/Interpreters/InterpreterSystemQuery.cpp

```diff
@@ -400,6 +403,96 @@ void InterpreterSystemQuery::restartReplicas(Context & system_context)
     pool.wait();
 }
 
+void InterpreterSystemQuery::dropReplicaByZKPath(ASTSystemQuery & query)
+{
+    context.checkAccess(AccessType::SYSTEM_DROP_REPLICA);
+    auto to_drop_path = query.replica_zk_path + "/replicas/" + query.replica;
+    auto & catalog = DatabaseCatalog::instance();
+    StorageReplicatedMergeTree::Status status;
+
+    for (auto & elem : catalog.getDatabases())
+    {
+        DatabasePtr & database = elem.second;
+        for (auto iterator = database->getTablesIterator(context); iterator->isValid(); iterator->next())
+        {
+            if (auto * storage_replicated = dynamic_cast<StorageReplicatedMergeTree *>(iterator->table().get()))
+            {
+                storage_replicated->getStatus(status);
+
+                if (status.replica_path.compare(query.replica_zk_path + "/replicas/" + status.replica_name) == 0)
```

Comment:
If local table with such `replica_path` exists and `DROP TABLE` should be used, then it's better to throw exception right here without call of `StorageReplicatedMergeTree::dropReplica(...)`